### PR TITLE
[issue-366] Support Retention feature

### PR DESCRIPTION
**Pull request by ManoharVanam**
Supporting data retention feature

Ref: https://github.com/HuaweiBigData/carbondata/issues/366

### CODE REVIEW

File: integration/spark/src/test/scala/org/carbondata/integration/spark/testsuite/dataretention/DataRetentionTestCase.scala [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        " IN ('china','ind','aus','eng') GROUP BY country"
+      ),
+      Seq(Row("aus", 9), Row("eng", 9), Row("ind", 9))
+    )
+  }
+
+  test("RetentionTest2") {
+    // delete ind, aus segments
+    sql("SHOW LOADS FOR TABLE DataRetentionTable").show()
+    sql("DELETE SEGMENTS FROM TABLE DataRetentionTable where STARTTIME before '" + time + "'")
```
Comment (ManoharVanam) [inline integration/spark/src/test/scala/org/carbondata/integration/spark/testsuite/dataretention/DataRetentionTestCase.scala]: please check issue - 366, I have written complete description

File: integration/spark/src/test/scala/org/carbondata/integration/spark/testsuite/dataretention/DataRetentionTestCase.scala [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      sql("SELECT country, count(salary) AS amount FROM DataRetentionTable WHERE country" +
+        " IN ('china','ind','aus','eng') GROUP BY country"
+      ),
+      Seq(Row("eng", 9))
+    )
+  }
+
+  test("RetentionTest3") {
+    // delete load 2 and load ind segment
+    sql("SHOW LOADS FOR TABLE DataRetentionTable").show()
```
Comment (ManoharVanam) [inline integration/spark/src/test/scala/org/carbondata/integration/spark/testsuite/dataretention/DataRetentionTestCase.scala]: please check issue - 366

File: integration/spark/src/test/scala/org/carbondata/integration/spark/testsuite/dataretention/DataRetentionTestCase.scala [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        "OPTIONS (PARTITIONER [PARTITION_COUNT=3])"
+    )
+
+    sql("LOAD DATA FACT FROM'" + resource + "dataretention1.csv' INTO CUBE DataRetentionTable " +
+      "OPTIONS(DELIMITER ',')")
+    sql("LOAD DATA FACT FROM'" + resource + "dataretention2.csv' INTO CUBE DataRetentionTable " +
+      "OPTIONS(DELIMITER ',')")
+
+    Thread.sleep(1000)
+    time = new SimpleDateFormat(CarbonCommonConstants.CARBON_TIMESTAMP_DEFAULT_FORMAT)
```
Comment (ManoharVanam) [inline integration/spark/src/test/scala/org/carbondata/integration/spark/testsuite/dataretention/DataRetentionTestCase.scala]: Iam using this variable while executing the second test case

File: integration/spark/src/test/scala/org/carbondata/integration/spark/testsuite/dataretention/DataRetentionTestCase.scala [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    .getCanonicalPath
+
+  val resource = currentDirectory + "/src/test/resources/"
+  var time = ""
+
+  override def beforeAll {
+    CarbonProperties.getInstance.addProperty(CarbonCommonConstants.MAX_QUERY_EXECUTION_TIME, "1")
+
+    sql(
+      "CREATE CUBE DataRetentionTable DIMENSIONS (ID Integer, date String, country String, name " +
```
Comment (ManoharVanam) [inline integration/spark/src/test/scala/org/carbondata/integration/spark/testsuite/dataretention/DataRetentionTestCase.scala]: ok

**Comment by gvramana** [comment]
raised and tracked as part of refer #432

**Pull request closed without merging**